src/api/api.ts

```diff
@@ -0,0 +1,57 @@
+import { Product } from '../types/Product';
+
+export const BASE_URL
+  = 'https://mate-academy.github.io/react_phone-catalog';
+
+const BASE_API_URL = `${BASE_URL}/_new/products`;
+
+export const getAllProducts = (): Promise<Product[]> => {
+  return fetch(`${BASE_API_URL}.json`)
+    .then(response => response.json());
+};
+
+export const getHotPriceProducts = () => {
+  return getAllProducts()
+    .then(products => products
```

Comment:
Well, basically, you fill never do filtering like this on a FE side and this will always be handled on the backend. 
However, as a simplification for training purposes this may work.